code/Equipment/Weapons/WeaponComponent.cs

```diff
@@ -160,24 +161,41 @@ public Vector3 GetStartPosition( bool isDroppable = false )
 		if ( FiringType is FiringType.Cursor )
 			return Vector3.Zero;
 
-		if ( isDroppable )
-			return Transform.Position.WithY( 0f );
-
 		if ( Equipment.Grub is not { } grub )
 			return Vector3.Zero;
 
+		var controller = grub.CharacterController;
+
+		if ( isDroppable )
+		{
+			// Perform a forward trace to find the position to drop the item
+			var dropTr = Scene.Trace.Ray( grub.EyePosition.Position, grub.EyePosition.Position + grub.Transform.Rotation.Forward * 25f )
+				.IgnoreGameObjectHierarchy( grub.GameObject )
+				.IgnoreGameObject( GameObject )
+				.Radius( 1f )
+				.Run();
+
+			var startPosition = dropTr.EndPosition;
+
+			if ( dropTr.Hit )
+			{
+				startPosition -= grub.Transform.Rotation.Forward * 10f;
+			}
+
+			return startPosition.WithY( 512 );
```

Comment:
Specify the float identifier `512f` when possible (also line 198)

Reply:
Fixed in next commit